cli/cmd/inject.go

```diff
@@ -188,6 +188,10 @@ func (rt resourceTransformerInject) transform(bytes []byte) ([]byte, []inject.Re
 	}
 
 	if len(rt.overrideAnnotations) > 0 {
+		// Do not set ingress annotation if its manual
+		if val, ok := rt.overrideAnnotations[k8s.ProxyInjectAnnotation]; ok && val == k8s.ProxyInjectIngress && rt.injectProxy {
+			delete(rt.overrideAnnotations, k8s.ProxyInjectAnnotation)
+		}
```

Comment:
You can move `delete(rt.overrideAnnotations, k8s.ProxyInjectAnnotation)` up under the existing `if injectProxy` block, and you don't need to check for the existence of the entry prior to deleting it :wink: